autocorpus/bioc_supplementary.py

```diff
@@ -0,0 +1,311 @@
+            inside_table = False
+            tables.append(table_lines)
+            table_lines = []
+            continue
+
+    for line in lines_to_remove:
+        temp_text_output.remove(line)
+
+    tables_output = []
+    # Remove lines that are just dashes (table separators)
+    for table in tables:
+        table = [line for line in table if not re.match(r"^\s*-+\s*$", line)]
+
+        # Extract rows from the identified table lines
+        rows = []
+        for line in table:
+            # Match only lines that look like table rows (contain pipes)
+            if re.search(r"\|", line):
+                # Split the line into cells using the pipe delimiter and strip whitespace
+                cells = [
+                    cell.strip()
+                    for cell in line.split("|")
+                    if not all(x in "|-" for x in cell)
+                ]
+                if cells:
+                    rows.append(cells)
+
+        # Determine the maximum number of columns in the table
+        num_columns = max(len(row) for row in rows)
+
+        # Pad rows with missing cells to ensure they all have the same length
+        for row in rows:
+            while len(row) < num_columns:
+                row.append("")
+
+        # Create a DataFrame from the rows
+        df = pd.DataFrame(rows[1:], columns=rows[0])
+        tables_output.append(df)
+    text_output = "\n\n".join(temp_text_output)
+    return text_output, tables_output
+
+
+def replace_unicode(text):
+    """Replaces specific Unicode characters with their corresponding replacements in the given text.
+
+    Args:
+        text (str or list): The input text or list of texts to process.
+
+    Returns:
+        str or list: The processed text or list of processed texts.
+
+    If the input `text` is empty or None, the function returns None.
+
+    If the input `text` is a list, it iterates over each element of the list and replaces the following Unicode characters:
+        - '\u00a0': Replaced with a space ' '
+        - '\u00ad': Replaced with a hyphen '-'
+        - '\u2010': Replaced with a hyphen '-'
+        - '\u00d7': Replaced with a lowercase 'x'
+
+    If the input `text` is not a list, it directly replaces the Unicode characters mentioned above.
+
+    Returns the processed text or list of processed texts.
+    """
+    if not text:
+        return None
+    if isinstance(text, list):
+        clean_texts = []
+        for t in text:
+            if t:
+                clean_texts.append(
+                    t.replace("\u00a0", " ")
+                    .replace("\u00ad", "-")
+                    .replace("\u2010", "-")
+                    .replace("\u00d7", "x")
+                )
+        return clean_texts
+    else:
+        clean_text = (
+            text.replace("\u00a0", " ")
+            .replace("\u00ad", "-")
+            .replace("\u2010", "-")
+            .replace("\u00d7", "x")
+        )
+        return clean_text
```

Comment:
I don't like this repetition... Maybe make a separate function to be called from this one that just handles a single string. That way there'll only be one place to update if you ever want to change the bevaviour